docs/csharp/whats-new/csharp-7.md

```diff
@@ -361,17 +361,20 @@ modified. The local variable has been declared with the `ref` modifier,
 and it will take a `ref` return. You must initialize a `ref` variable when
 it is declared, you cannot split the declaration and the initialization.
 
-The C# language has two other rules that protect you from misusing
+The C# language has three other rules that protect you from misusing
 the `ref` locals and returns:
 
 * You cannot assign a value to a `ref` variable.
     - That disallows statements like `ref int i = sequence.Count();`
 * You cannot return a `ref` to a variable whose lifetime does not extend beyond the execution of the method.
-    - That means you cannot return a reference to a local variable, or similar scope.
+    - That means you cannot return a reference to a local variable or a variable with a similar scope.
+* You cannot return a `ref` to a variable that doesn't exist when the method is ready to set or return the reference.
```

Comment:
@rpetrusha I'm attempting to match the structure of the prior rule, but I fear that the following doesn't make sense out-of-context (i.e., without making note that this rule deals exclusively with the possibility of using `ref` locals and returns with async methods).

> You cannot return a `ref` to a variable when there is no way of knowing if its final value has been set. [should it add "... when the method is prepared to return the reference" or "... when the method returns"?]

--> followed by the change in language you suggest below ...

> That means you cannot use `ref` locals and returns with async methods because the compiler cannot know if the referenced variable has been set to its final value when the async method returns.

The alternative is to break from rule 2's format and go with something like ...

* `ref` locals and returns can't be used with async methods.
  - The compiler cannot know if the referenced variable has been set to its final value when the async method returns.